client/src/app/deck/deck.tsx

```diff
@@ -86,21 +90,88 @@ const Deck = ({setMenuSection}: Props) => {
 		setMenuSection('mainmenu')
 	}
 
+	const loadSavedDecks = () => {
+		let lskey, deck
+		const deckList = []
+
+		//loop through Local Storage keys
+		for (let i = 0; i < localStorage.length; i++) {
+			lskey = localStorage.key(i)
+			deck = lskey?.replace(/Loadout_/g, '')
+
+			//if ls key contains 'Loadout_' then add to deckList array.
+			if (lskey?.includes('Loadout_')) {
+				deckList.push(deck)
+			}
+		}
+
+		console.log(
+			'Loaded ' + deckList.length + ' decks from Local Storage',
+			deckList.sort()
+		)
+		setLoadedDecks(deckList.sort())
+	}
+
+	useEffect(() => {
+		loadSavedDecks()
+	}, [])
+
 	const clearDeck = () => {
 		setPickedCards([])
 	}
+
 	const saveDeck = () => {
-		localStorage.setItem('Loadout_' + deckName, JSON.stringify(pickedCards))
+		// Check if deckName is a valid string
+		if (!deckName || /^\s*$/.test(deckName)) {
+			alert('Invalid deck name. Please try again.')
+			return
+		}
+		const newDeckName = deckName.trim()
+
+		// Check if deck name already exists
+		if (loadedDecks.includes(newDeckName)) {
+			const confirmOverwrite = confirm(
+				'"' + newDeckName + '" already exists! Would you like to overwrite it?'
+			)
+			if (!confirmOverwrite) return
+			localStorage.removeItem('Loadout_' + newDeckName)
+			setLoadedDecks([...loadedDecks].filter((d) => d !== newDeckName))
+		}
+
+		// Save deck to Local Storage
+		localStorage.setItem('Loadout_' + newDeckName, JSON.stringify(pickedCards))
 		console.log(JSON.stringify(pickedCards))
+		setLoadedDecks([newDeckName, ...loadedDecks])
+		loadSavedDecks()
+		alert('"' + newDeckName + '" was saved to Local Storage!')
 	}
-	const loadDeck = () => {
-		const deck = localStorage.getItem('Loadout_' + deckName)
-		if (!deck) return
+
+	const loadDeck = (selectedDeck: any) => {
+		console.log('Loading deck: ', selectedDeck)
+		setSelectedDeck(selectedDeck)
+		if (!selectedDeck) return console.log('Could not load deck...')
+		const deck: any = localStorage.getItem('Loadout_' + selectedDeck)
 		const deckIds = JSON.parse(deck).filter(
 			(card: CardT) => TYPED_CARDS[card.cardId]
 		)
 		setPickedCards(deckIds)
 	}
+
+	const deleteDeck = (deck: string) => {
+		const confirmDelete = confirm(
```

Comment:
Same thigns as alert, these dialogs are deprecated and generally shouldn't be use. For the confirm we can use custom model (e.g. similar as for the Imports/Exports)